src/Database/Relations/BelongsToMany.php

```diff
@@ -406,7 +406,51 @@ public function getOtherKey()
      */
     public function getRelatedIds($sessionKey = null)
     {
-        traceLog('Method BelongsToMany::allRelatedIds has been deprecated, use BelongsToMany::allRelatedIds instead.');
+        traceLog('Method BelongsToMany::getRelatedIds has been deprecated, use BelongsToMany::allRelatedIds instead.');
         return $this->allRelatedIds($sessionKey)->all();
     }
+
+    /**
+     * Sync the intermediate tables with a list of IDs or collection of models.
+     *
+     * @param  \Illuminate\Support\Collection|\Illuminate\Database\Eloquent\Model|array  $ids
+     * @param  bool  $detaching
+     * @return array
+     */
+    public function sync($ids, $detaching = true)
+    {
+        $changes = [
+            'attached' => [], 'detached' => [], 'updated' => [],
+        ];
+
+        // Get all the IDs for the related model
+        $current = $this->allRelatedIds()->all();
```

Comment:
Wouldn't a better fix be to adjust the getCurrentlyAttachedPivots() method to return the correct query builder instead? That way we don't have to override this method at all?

Reply:
You may be right, although I am not sure of all the implications that would have on the other classes using the `InteractsWithPivotTable` trait.

Reply:
Presumably it would fix similar issues with those method consumers at the same time. @bennothommo any thoughts? (btw, Ben's got spotty internet right now, might take a couple days)

Reply:
It's definitely worth investigating if I have some help from Ben or yourself...

Reply:
The new commit addresses the issue with the same result, but replaces getCurrentlyAttachedPivots() method instead of replacing the sync() method.

Note that it currently only affects `BelongToMany` relation but not `MorphMany`.